### ISSUE TITLE
[@nativescript/social-share] not working on ipad

### ISSUE THREAD

**Issue by vallemar** [NativeScript/plugins#79]
Sharing does not work on ipad, the ios version I run on is: 14.4

**Comment by vallemar** [comment]
I have gone to the `index.ios.js `file and I have copied the piece that I needed, adding a `CGRect` the problem has been solved. The change made has been this:

```
            if (presentViewController) {
                const page = Frame.topmost().currentPage;
                if (page && page.ios.navigationItem.rightBarButtonItems && page.ios.navigationItem.rightBarButtonItems.count > 0) {
                    presentViewController.barButtonItem = page.ios.navigationItem.rightBarButtonItems[0];
                }
                else {
                    presentViewController.sourceView = page.ios.view;
                }
            }
```
to
```
            if (presentViewController) {
                const page = Frame.topmost().currentPage;
                if (page && page.ios.navigationItem.rightBarButtonItems && page.ios.navigationItem.rightBarButtonItems.count > 0) {
                    presentViewController.barButtonItem = page.ios.navigationItem.rightBarButtonItems[0];
                }
                else {
                    presentViewController.sourceView = page.ios.view;
  ->              presentViewController.sourceRect = CGRectMake(CGRectGetMidX(page.ios.view.bounds), CGRectGetMaxY(page.ios.view.bounds), 0,0)
  ->              presentViewController.permittedArrowDirections = UIPopoverArrowDirection.Unknown
                }
            }
```

I do not know if it is the best solution but while it is corrected I will keep this piece in my code.

**Comment by byGui** [comment]
I have the same problem on iPad with iOS v14.4 and this fix works for me

**Comment by NathanWalker** [comment]
Thanks @vallemar great solution. Published in `2.0.4` now 👍
